### PR TITLE
Make encryption default if Security is given arguments

### LINKED ISSUE #3886: Default value for Security(require_encryption=) keyword
I'm curious about the `require_encryption=` keyword.  Currently it defaults to False.

I'm curious if it should default to True if the user provides TLS information

cc @jcrist

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #3886

### CODE REVIEW

**Comment by mrocklin** [comment]
Merging tomorrow if there are no comments.  cc @jcrist

File: distributed/security.py [hunk 1]
```diff
@@ -60,13 +60,15 @@ class Security:
...
         extra = set(kwargs).difference(self.__slots__)
         if extra:
             raise TypeError("Unknown parameters: %r" % sorted(extra))
-        self._set_field(
-            kwargs, "require_encryption", "distributed.comm.require-encryption"
-        )
+        if require_encryption is None:
+            require_encryption = dask.config.get("distributed.comm.require-encryption")
+        if require_encryption is None:
+            require_encryption = not not kwargs
```
Comment (samaust) [inline distributed/security.py:70]: Can you explain what this line does?

Comment (quasiben) [reply distributed/security.py:70]: I suspect this is a mistake ? `not not` ?

Comment (mrocklin) [reply distributed/security.py:70]: `not not` turns anything into True/False if it is truthy/falsey.

```python

In [1]: not not {"x": 1}                                                                                                                                                                                          
Out[1]: True

In [2]: not not {}                                                                                                                                                                                                
Out[2]: False
```

Comment (mrocklin) [reply distributed/security.py:70]: Probably calling `bool` is better

Comment (quasiben) [reply distributed/security.py:70]: Oh!  Cool!  And we do that in other places throughout the dask ecosystem.  

So @samaust this line is saying any time the user is setting security settings (in kwargs) dask will now enable `require_encryption` without the user explicitly setting it.  Essentially the title of the PR.

**Comment by quasiben** [comment]
merging in

**Pull request merged**